objectRecognitionSource/src/main/java/edu/ml/tensorflow/ObjectDetector.java

```diff
@@ -74,12 +81,35 @@
      * @param vantiq        The Vantiq variable used to connect to the VANTIQ SDK. Either authenticated, or set to null.
      * @param sourceName    The name of the VANTIQ Source
      */
-    public ObjectDetector(float thresh, String graphFile, String labelFile, double[] anchorArray, ImageUtil imageUtil, String outputDir, 
+    public ObjectDetector(float thresh, String graphFile, String labelFile, String metaFile, double[] anchorArray, ImageUtil imageUtil, String outputDir, 
             Boolean labelImage, int saveRate, Vantiq vantiq, String sourceName) {
         try {
             GRAPH_DEF = IOUtil.readAllBytesOrExit(graphFile);
-            LABELS = IOUtil.readAllLinesOrExit(labelFile);
-            this.anchorArray = anchorArray;
+            // Parse meta file if it exists, and either the label file or anchor config options have not been set
+            if (metaFile != null && (labelFile == null || anchorArray == null)) {
+                parseMetaFile(metaFile);
+            }
+            // If label file exists, use it. Otherwise, use the meta file's labels.
+            if (labelFile != null) {
+                LABELS = IOUtil.readAllLinesOrExit(labelFile);
+            } else {
+                LABELS = (List<String>) metaFileMap.get("labels");
```

Comment:
At this point it should be non-null. If both the meta file and label file were non-null, we would not make it to this point. Otherwise, the meta file would be non-null and the label file would be null, which would trigger the previous condition and call parseMetaFile.

Let me know if I'm missing something here, or if I should still add a check.

Reply:
As I read the code, it says

```
                    if (metaFile != null && (labelFile == null || anchorArray == null)) {
	                parseMetaFile(metaFile);
	            }
	            // If label file exists, use it. Otherwise, use the meta file's labels.
	            if (labelFile != null) {
	                LABELS = IOUtil.readAllLinesOrExit(labelFile);
	            } else {
	                LABELS = (List<String>) metaFileMap.get("labels");
                    }
```
it says, if the metafile is there & some other stuff is true, parse the meta file (which will create metaFileMap).

Then, regardless of whether we did that, if there's no label file, read the labelFile from the metaFileMap.

So, if we have neither a labelFile nor a metaMap, we'll blow up here.  Now, maybe someone higher up verifies that at least one of these is true, but that dependency is "far way" and pretty-well buried (down in YoloProcessor.setup()).  Personally, I'd add a check.  

Also, I noticed (didn't before) & added a comment that we should add parse any metaFile we get -- no reason not to.  Currently, I think we don't if we have a metaFile & we have both labels & anchors.  In the future, we'll use more from there, so we should verify the metaFile that they give us, I think.

Reply:
Ok so I just added the fix to parse the meta file whenever we get it. 

And you're exactly right, the check is in YoloProcessor.setup() which says that if we don't have either a label or a meta file, then we will throw an exception before initializing the ObjectDetector class. In any case, I'll add the check here just to be completely bullet proof.